### PR TITLE
slave node uses systemd

### PR DESCRIPTION
**Pull request by atarkowska**
This PR improves slave node using systemd

To test it:
- deploy jenkins (use https://github.com/aleksandra-tarkowska/omero-ssh-systemd-jenkins-slave)
- attache to jenkins container to find out `env | grep JENKINS_PORT_8080_TCP`
- build using `make build`
- start container:
  
  ```
  UNIX: make start ENV="-e JENKINS_PORT_8080_TCP_ADDR=172.17.0.2 -e JENKINS_PORT_8080_TCP_PORT=8080" VOLUME="-v /sys/fs/cgroup:/sys/fs/cgroup:ro -v /run"
  ```
  
  ```
  OSX: make start PORTS="--privileged" ENV="-e JENKINS_PORT_8080_TCP_ADDR=172.17.0.2 -e JENKINS_PORT_8080_TCP_PORT=8080"
  ```
- jenkins logs will show:
  
  ```
  jenkins_1 | INFO: Accepted connection #2 from /172.17.0.4:43378
  ```
- attache to devslave:
  
  ```
  $ docker exec -it devslave bash -c "service jenkins status"
  service jenkins status                                                                                    Redirecting to /bin/systemctl status  jenkins.service    
  ● jenkins.service - Jenkins slave
  Loaded: loaded (/etc/systemd/system/jenkins.service; enabled; vendor preset: disabled)
  Active: active (running) since Thu 2016-04-07 21:14:02 UTC; 14min ago
   Docs: https://developer.cloudbees.com/bin/view/DEV/On-Premise+Executors
  Main PID: 223 (java)
   CGroup: /system.slice/jenkins.service
           └─223 java -jar /tmp/swarm-client-2.0-jar-with-dependencies.jar -fsroot /home/omero -master http://172.17.0.2:8080 -lab...
  
  Apr 07 21:14:02 cca03c032f77 jenkins-slave.sh[223]: Apr 07, 2016 9:14:02 PM hudson.remoting.jnlp.Main$CuiListener <init>
  Apr 07 21:14:02 cca03c032f77 jenkins-slave.sh[223]: INFO: Jenkins agent is running in headless mode.
  Apr 07 21:14:02 cca03c032f77 jenkins-slave.sh[223]: Apr 07, 2016 9:14:02 PM hudson.remoting.jnlp.Main$CuiListener status
  Apr 07 21:14:02 cca03c032f77 jenkins-slave.sh[223]: INFO: Locating server among [http://172.17.0.2:8080/]
  Apr 07 21:14:02 cca03c032f77 jenkins-slave.sh[223]: Apr 07, 2016 9:14:02 PM hudson.remoting.jnlp.Main$CuiListener status
  Apr 07 21:14:02 cca03c032f77 jenkins-slave.sh[223]: INFO: Connecting to 172.17.0.2:50000
  Apr 07 21:14:02 cca03c032f77 jenkins-slave.sh[223]: Apr 07, 2016 9:14:02 PM hudson.remoting.jnlp.Main$CuiListener status
  Apr 07 21:14:02 cca03c032f77 jenkins-slave.sh[223]: INFO: Handshaking
  Apr 07 21:14:02 cca03c032f77 jenkins-slave.sh[223]: Apr 07, 2016 9:14:02 PM hudson.remoting.jnlp.Main$CuiListener status
  Apr 07 21:14:02 cca03c032f77 jenkins-slave.sh[223]: INFO: Connected
  ```
- check jenkins http://jenkins:8080/computer/ if new node is up

cc: @joshmoore

### CODE REVIEW

**Comment by joshmoore** [comment]
My concern here, @aleksandra-tarkowska, is that this strategy makes it difficult to deploy to hub. If we can have only the changes here for devslave, then we can have a separate devjenkins (devmaster?) repo, and each of those can produce an image on hub which you can re-use in docker-compose.

**Comment by atarkowska** [comment]
np, I have just rebased with slave only.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,27 @@
+NS = openmicroscopy
+VERSION ?= latest
```
Comment (atarkowska) [inline Makefile]: change to local

File: Dockerfile [hunk 2]
```diff
@@ -1,33 +1,49 @@
 FROM openmicroscopy/omero-ssh-systemd
+
 MAINTAINER ome-devel@lists.openmicroscopy.org.uk
 
-ENV JENKINS_SWARM_VERSION 1.24
 ENV LANG en_US.UTF-8
-ARG JAVAVER=openjdk18
+ENV JENKINS_LABEL /etc/alternatives/java_sdk
```
Comment (joshmoore) [inline Dockerfile]: Drop

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+REPO = devslave-c7-docker
```
Comment (joshmoore) [inline Makefile]: drop "-docker"

File: .travis.yml [hunk 4]
```diff
@@ -1,16 +1,37 @@
...
-    - docker build --build-arg JAVAVER=$JAVAVER -t devslave-c7-test .
+  - docker run -d --name jenkins jenkins
+  - docker inspect -f {{.State.Running}} jenkins
+  - d=10; while ! docker logs jenkins 2>&1 | grep "Jenkins is fully up and running" ; do sleep 10; d=$[$d -1]; if [ $d -lt 0 ]; then exit 1; fi; done
+  - JENKINS_ADDR=`docker inspect --format '{{ .NetworkSettings.IPAddress }}' jenkins`
+  - JENKINS_PORT=50000
+  - make start ENV="-e JENKINS_PORT_8080_TCP_ADDR=$JENKINS_ADDR -e JENKINS_PORT_8080_TCP_PORT=$JENKINS_PORT" VOLUMES="-v /sys/fs/cgroup:/sys/fs/cgroup:ro -v /run"
+  - docker inspect -f {{.State.Running}} devslave
+  - SLAVE_ADDR=`docker inspect --format '{{ .NetworkSettings.IPAddress }}' devslave`
+  - d=10; while ! docker logs jenkins 2>&1 | grep "from /${SLAVE_ADDR}" ; do sleep 10; d=$[$d -1]; if [ $d -lt 0 ]; then exit 1; fi; done
```
Comment (joshmoore) [inline .travis.yml:33]: Very nice!

**Comment by joshmoore** [comment]
Nice. Merging. To be "integration tested" in the devspace along with @aleksandra-tarkowska's upcoming redis PR. This should significantly speed up the build.

**Pull request merged**